scs2-simulation/src/main/java/us/ihmc/scs2/simulation/physicsEngine/YoMatrix.java

```diff
@@ -36,11 +63,46 @@ public YoMatrix(String name, int maxNumberOfRows, int maxNumberOfColumns, YoRegi
       {
          for (int column = 0; column < maxNumberOfColumns; column++)
          {
-            variables[row][column] = new YoDouble(name + "_" + row + "_" + column, registry);
+            switch (checkNames(rowNames, columnNames))
+            {
+               case NONE:
+               {
+                  variables[row][column] = new YoDouble(name + "_" + row + "_" + column, description, registry);  // names are simply the row and column indices
+                  break;
+               }
+               case ROWS:
+               {
+                  if (maxNumberOfColumns > 1)
+                     throw new IllegalArgumentException("The YoMatrix must be a column vector if only row names are provided, else unique names cannot be generated.");
+
+                  variables[row][column] = new YoDouble(name + "_" + rowNames[row], description, registry);  // names are the row names, no column identifier
+                  break;
+               }
+               case ROWS_AND_COLUMNS:
+               {
+                  variables[row][column] = new YoDouble(name + "_" + rowNames[row] + "_" + columnNames[column], description, registry);  // names are the row and column names
+                  break;
+               }
+            }
```

Comment:
The new constructors for `YoMatrix` with a description give the same description to each `YoDouble` in the matrix. I can't  imagine us needing such fine-grained control as to give each entry a unique description.